File: instrumentation/newrelic-scala-zio-api/src/main/java/com/newrelic/zio/api/TraceOps_Instrumentation.java
package com.newrelic.zio.api;

import com.newrelic.api.agent.weaver.Weave;
import com.newrelic.api.agent.weaver.Weaver;
import zio.ZIO;

@Weave(originalName = "com.newrelic.zio.api.TraceOps$")
public class TraceOps_Instrumentation {
  public <R, E, A> ZIO<R, E, A> txn(final ZIO<R, E, A> body) {
    return Util.wrapTrace(Weaver.callOriginal());
  }
}

File: instrumentation/scala-monix-api/src/main/scala/com/newrelic/monix/api/TraceOps_Instrumentation.scala
package com.newrelic.monix.api

import com.newrelic.api.agent.weaver.Weaver
import com.newrelic.api.agent.weaver.scala.{ScalaMatchType, ScalaWeave}
import monix.eval.Task


@ScalaWeave(`type` = ScalaMatchType.Object, `originalName` = "com.newrelic.monix.api.TraceOps")
class TraceOps_Instrumentation {
  def txn[A](body: Task[A]): Task[A] = Util.wrapTrace(Weaver.callOriginal)
}
